src/main/java/org/sonarsource/auth/github/GitHubIdentityProvider.java

```diff
@@ -124,6 +137,54 @@ private GsonUser getUser(OAuthService scribe, Token accessToken) {
     return GsonTeams.parse(responseBody);
   }
 
+  public boolean isOrganizationMembershipRequired() { return settings.organizations().length > 0; }
+
+  private boolean isOrganizationsMember(Token accessToken, String login) {
+    for (String organization : settings.organizations()) {
+      if (isOrganizationMember(accessToken, organization, login)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  private boolean isUnauthorized(Token accessToken, String login) {
+    return isOrganizationMembershipRequired() && !isOrganizationsMember(accessToken, login);
+  }
+
+  /**
+   * Check to see that login is a member of organization.
+   *
+   * A 204 response code indicates organization membership.  302 and 404 codes are not treated as exceptional,
```

Comment:
I'm thinking about the 404 code => It means that the user enter an organization that doesn't exist, that's it ? It this case, we should fail with an explicit message saying that the organization doesn't exist.